examples/convert_file_formats.py

```diff
@@ -0,0 +1,169 @@
+    "back_L": "hipL",
+    "back_R": "hipR",
+    "abdomen_post": "tail",
+}
+
+
+# %%
+# Now we can run the function and see that
+# the keypoints have been renamed.
+# this function takes the dataset and the rename_dict as input.
+def rename_keypoints(ds, rename_dict):
+    # get the current names of the keypoints
+    keypoint_names = ds.coords["keypoints"].values
+    print("Original keypoints:", keypoint_names)
+    # rename the keypoints
+    if rename_dict is None or len(rename_dict) == 0:
+        print("No KPs to rename. Skipping renaming step.")
+    else:
+        new_keypoints = [rename_dict.get(kp, kp) for kp in keypoint_names]
+        print("New keypoints:", new_keypoints)
+        # Assign the modified values back to the Dataset
+        ds = ds.assign_coords(keypoints=new_keypoints)
+    return ds
+
+
+# %%
+# Delete Keypoints of your choice
+# --------------------------------
+# First, create a list of keypoints.
+# to delete modify this list accordingly
+
+kps_to_delete = ["abdomen_pre", "abdomen", "tailbase", "front_L", "front_R"]
+
+
+# %%
+# Now we can go ahead and delete those Keypoints
+def delete_keypoints(ds, delete_keypoints):
+    if delete_keypoints is None or len(delete_keypoints) == 0:
+        print("No KPs to delete. Skipping deleting step.")
+    else:
+        # Delete the specified keypoints
+        # and their corresponding data
+        ds = ds.drop_sel(keypoints=delete_keypoints)
+    return ds
+
+
+# %%
+# Delete Keypoints of your choice
+# --------------------------------
+# Again create a list with the
+# desired order of the keypoints
+
+ordered_keypoints = ["nose", "earL", "earR", "neck", "hipL", "hipR", "tail"]
+
+
+# %%
+# Now we can go ahead and reorder
+# those keypoints
+def reorder_keypoints(ds, ordered_keypoints):
+    # reorder the keypoints
+    if ordered_keypoints is None or len(ordered_keypoints) == 0:
+        print("No KPs to reorder. Skipping reordering step.")
+    else:
+        # Reorder the keypoints in the Dataset
+        ds = ds.reindex(keypoints=ordered_keypoints)
+    return ds
+
+
+# %%
+# and how can I use this when I only have my filepath????
+# -----------
+# Great! Now we know how we can modify our movement datasets!
+# let's put it all together and see how we could use this in a real world
+# scenario.
+# This function will convert all files in the folder to the desired format.
+# The function will rename the keypoints, delete specified keypoints,
+# and reorder them in the dataset.
+# The function will then save the modified dataset to a new file.
+
+
+FPATH = "/path/to/your/data/"
+
+
+def convert_all(FPATH, ext="_inference.slp"):
+    source_folder = pathlib.Path(FPATH)
+    fpaths = list(source_folder.rglob(f"*{ext}"))
+
+    for fpath in fpaths:
+        fpath = pathlib.Path(fpath)
+
+        # this determines the filename of your modified file
+        # change it if you like to change the filename
+        dest_path = fpath.parent / "tracking_2D_8KP.csv"
```

Comment:
I'm confused by this, won't all the output files end up with the same filename then (essentially overwriting at each iteration)?

Reply:
How I would handle this:
- pass a `source_dir` and a `target_dir` separately as arguments to the function
- the converted files should be written to `target_dir` and their filename can be determined based on the source filename + some additional suffix to indicate the change, for example  sth like `f"{fpath.stem}_dlc.csv"`

This is not they only way, but it keeps things neat.
But I'm fine with other ways of doing it, as long as nothing get overwritten.